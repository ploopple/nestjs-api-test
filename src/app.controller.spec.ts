import * as request from 'supertest';
import * as rp from 'request-promise';
import { Test } from '@nestjs/testing';
import { TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';

import { AppModule } from './app.module';

const API =
  'https://3334-ploopple-nestjsapitest-0whk9u5keso.ws-eu54.gitpod.io/graphql';
describe('appController', () => {
  let app: INestApplication;
  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });
  it(`/GET cats`, () => {
    return request(app.getHttpServer()).get('/').expect(200).expect('Hello');
  });

  it("should be able to see author's profile without sensitive info being displayed", async () => {
    const query = `
    query {
        getAllUsers{
          name
        }
      }
        `;

    const { data } = await rp({
      method: 'POST',
      uri: API,
      body: { query },
      json: true,
    });
    expect(data).toMatchSnapshot();
  });
});
