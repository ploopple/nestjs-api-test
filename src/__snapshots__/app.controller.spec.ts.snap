// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appController should be able to see author's profile without sensitive info being displayed 1`] = `
Object {
  "getAllUsers": Array [
    Object {
      "name": "Mohamed",
    },
    Object {
      "name": "Banen",
    },
    Object {
      "name": "Ayoub",
    },
    Object {
      "name": "Faysal",
    },
    Object {
      "name": "Mohamed",
    },
    Object {
      "name": "Banen",
    },
    Object {
      "name": "Ayoub",
    },
    Object {
      "name": "Faysal",
    },
    Object {
      "name": "Mohamed",
    },
    Object {
      "name": "Banen",
    },
    Object {
      "name": "Ayoub",
    },
    Object {
      "name": "Faysal",
    },
  ],
}
`;
